### DISCUSSION TITLE
[Todo App: API com ASP.NET Core, CQRS e EF Core] [Executando as migrações] O pacote 'Microsoft.EntityFrameworkCore.Design' é incompatível com as estruturas 'all' no projeto

### DISCUSSION THREAD

**Discussion by fecassaDev** [balta-io/forum#421] [Tira Dúvidas] [answered]
Fala pessoal, estou tentando replicar o código utilizando donet 7.0.402 mas não estou conseguindo fazer as migrations. 
Após executar o comando **ef migrations add** (da forma exibida no curso) , recebo uma mensagem dizendo que não possuo o pacote Microsoft.EntityFrameworkCore.Design no meu projeto de api.

Mas ao tentar adiconar este pacote ao projeto da api recebo a seguinte mensagem: "**error: O pacote 'Microsoft.EntityFrameworkCore.Design' é incompatível com as estruturas 'all' no projeto...**"

Alguém já passou por isto?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @fecassaDev o pacote deve ser adicionado no projeto API, que deve estar na mesma versão do .NET (7 no seu caso).
Se você referencia projetos de versões menores, talvez também tenha problemas, então, o ideal é manter todos os projetos no .NET 7.

**Comment by fecassaDev** [comment]
Funcionou, obrigado
